[error] failed to query source: unsupported query type

### Nomad Autoscaler version
Nomad Autoscaler v0.4.9 (82611d2)
Nomad v1.11.3

### Operating system and Environment details
Linux HashiCarp 6.12.73-v8+ #1947 SMP PREEMPT Thu Feb 19 12:31:05 GMT 2026 aarch64 GNU/Linux
Docker Version: 29.3.0

### Issue
A simple policy that worked on previous version now fails with an error message "failed to query source: unsupported query type".

### Reproduction steps
Run a nomad job with the following scaling block and Nomad Autoscaler will report errors.
```
    scaling {
      enabled = true
      min     = 1
      max     = 10

      policy {
        check "fixed-value-check" {
          strategy "fixed-value" {
            value = 2
          }
        }
      }
    }
```

#### Expected Result
Expected policy to evaluate successfully. (as previous versions did.)

#### Actual Result
```
[ERROR] policy_manager: encountered an error monitoring policy IDs: error="handler: unable to execute policy ID: 9851171e-7760-4d02-b792-632401bca43b, failed get count from metrics: failed to query source: failed to query source: unsupported query type \"\""
```

### Job file (if appropriate)
```
job "webapp" {
  datacenters = ["dc1"]

  group "demo" {
    network {
      mode = "host"
      port "http" {
      }
    }

    service {
      port = "http"
      name = "webapp"
      tags = ["urlprefix-webapp.service.consul/"]
      check {
        port     = "http"
        type     = "http"
        path     = "/"
        method   = "HEAD"
        interval = "10s"
        timeout  = "2s"
      }
    }

    task "app" {
      driver = "docker"

      config {
        image = "hashicorp/http-echo:1.0"
        ports = ["http"]
        args = [
          "-listen", ":${NOMAD_PORT_http}",
        ]
      }
    }
    scaling {
      enabled = true
      min     = 1
      max     = 10

      policy {
        check "fixed-value-check" {
          strategy "fixed-value" {
            value = 2
          }
        }
      }
    }
  }
}
```

### Nomad Autoscaler logs
```
2026-03-22T23:04:14.230Z [INFO]  agent: starting Nomad Autoscaler agent
2026-03-22T23:04:14.231Z [INFO]  agent: Nomad Autoscaler agent configuration:
2026-03-22T23:04:14.231Z [INFO]  agent
2026-03-22T23:04:14.231Z [INFO]  agent:         Bind Addrs: 0.0.0.0
2026-03-22T23:04:14.231Z [INFO]  agent:  High Availability: true
2026-03-22T23:04:14.231Z [INFO]  agent:          Log Level: info
2026-03-22T23:04:14.231Z [INFO]  agent:            Plugins: //plugins
2026-03-22T23:04:14.231Z [INFO]  agent:           Policies: 
2026-03-22T23:04:14.231Z [INFO]  agent:            Version: v0.4.9 (82611d2)
2026-03-22T23:04:14.231Z [INFO]  agent
2026-03-22T23:04:14.231Z [INFO]  agent: Nomad Autoscaler agent started! Log data will stream in below
2026-03-22T23:04:14.249Z [INFO]  agent: running in HA mode: lock_namespace=default lock_path=nomad/jobs/autoscaler/nomad/lock lock_ttl=1m0s lock_delay=30s
2026-03-22T23:04:14.249Z [INFO]  agent.http_server: server now listening for connections: address=0.0.0.0:27729
2026-03-22T23:04:14.258Z [INFO]  agent.plugin_manager: successfully launched and dispensed plugin: plugin_name=nomad-apm
2026-03-22T23:04:14.258Z [INFO]  agent.plugin_manager: successfully launched and dispensed plugin: plugin_name=prometheus
2026-03-22T23:04:14.258Z [INFO]  agent.plugin_manager: successfully launched and dispensed plugin: plugin_name=fixed-value
2026-03-22T23:04:14.258Z [INFO]  agent.plugin_manager: successfully launched and dispensed plugin: plugin_name=pass-through
2026-03-22T23:04:14.258Z [INFO]  agent.plugin_manager: successfully launched and dispensed plugin: plugin_name=target-value
2026-03-22T23:04:14.258Z [INFO]  agent.plugin_manager: successfully launched and dispensed plugin: plugin_name=threshold
2026-03-22T23:04:14.258Z [INFO]  agent.plugin_manager: successfully launched and dispensed plugin: plugin_name=nomad-target
2026-03-22T23:04:14.260Z [INFO]  policy_manager: starting policy source: source=nomad
2026-03-22T23:04:14.271Z [ERROR] policy_manager: encountered an error getting the latest version for policy: policyID=8a7f3f1c-068b-8c26-d78d-0c9082ad8aab
  error=
  | 3 errors occurred:
  | \t* policy validation failed
  | \t* empty policy, this policy won't execute any verification or scaling and should have enabled set to false
  | \t* empty checks, this policy won't execute any verification or scaling and should have enabled set to false
  | 
  
2026-03-22T23:04:14.275Z [ERROR] policy_manager: encountered an error getting the latest version for policy: policyID=57fe005f-87a9-8671-eba5-0d66f2efbe68
  error=
  | 3 errors occurred:
  | \t* policy validation failed
  | \t* empty policy, this policy won't execute any verification or scaling and should have enabled set to false
  | \t* empty checks, this policy won't execute any verification or scaling and should have enabled set to false
  | 
  
2026-03-22T23:04:29.320Z [ERROR] policy_manager: encountered an error monitoring policy IDs: error="handler: unable to execute policy ID: 9851171e-7760-4d02-b792-632401bca43b, failed get count from metrics: failed to query source: failed to query source: unsupported query type \"\""
2026-03-22T23:04:44.320Z [ERROR] policy_manager: encountered an error monitoring policy IDs: error="handler: unable to execute policy ID: 9851171e-7760-4d02-b792-632401bca43b, failed get count from metrics: failed to query source: failed to query source: unsupported query type \"\""
2026-03-22T23:04:59.319Z [ERROR] policy_manager: encountered an error monitoring policy IDs: error="handler: unable to execute policy ID: 9851171e-7760-4d02-b792-632401bca43b, failed get count from metrics: failed to query source: failed to query source: unsupported query type \"\""
2026-03-22T23:05:14.320Z [ERROR] policy_manager: encountered an error monitoring policy IDs: error="handler: unable to execute policy ID: 9851171e-7760-4d02-b792-632401bca43b, failed get count from metrics: failed to query source: failed to query source: unsupported query type \"\""

```

```
$ nomad scaling policy info 9851171e-7760-4d02-b792-632401bca43b
ID      = 17a7468d
Enabled = true
Target  = Namespace:default,Job:mirror,Group:mirror
Min     = 1
Max     = 10

Policy:
{
    "check": [
        {
            "fixed-value-check": [
                {
                    "strategy": [
                        {
                            "fixed-value": [
                                {
                                    "value": 2.0
                                }
                            ]
                        }
                    ]
                }
            ]
        }
    ]
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,9 @@ IMPROVEMENTS:
 * policy: Reuse identical APM query results within a single policy evaluation to avoid duplicate source requests for checks that share the same source, query, query window, and query window offset. [[GH-1252](https://github.com/hashicorp/nomad-autoscaler/pull/1252)] 
 * plugins: Fixed AWS-ASG plugin error handling so the underlying error is preserved when retry limit is reached. [[GH-1266](https://github.com/hashicorp/nomad-autoscaler/pull/1266)]
 
+BUG FIXES:
+* policy: Fixed-value strategy checks no longer depend on APM lookup or metric query results, allowing fixed-value scaling to proceed when the configured metrics source is unavailable. [[GH-1281](https://github.com/hashicorp/nomad-autoscaler/pull/1281)]
+
 ## 0.4.9 (January 6, 2026)
 
 BUG FIXES:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,7 @@ IMPROVEMENTS:
 
 BUG FIXES:
 * policy: Fixed a bug where misconfigured strategy checks could crash the autoscaler instead of following normal `on_error` and `on_check_error` behavior. [[GH-1275](https://github.com/hashicorp/nomad-autoscaler/pull/1275)]
-* policy: Fixed-value strategy checks no longer depend on APM lookup or metric query results, allowing fixed-value scaling to proceed when the configured metrics source is unavailable. [[GH-1281](https://github.com/hashicorp/nomad-autoscaler/pull/1281)]
+* policy: Fixed-value strategy checks no longer require APM source or query configuration. [[GH-1281](https://github.com/hashicorp/nomad-autoscaler/pull/1281)]
 
 ## 0.4.9 (January 6, 2026)
 

--- a/policy/check.go
+++ b/policy/check.go
@@ -12,6 +12,7 @@ import (
 
 	hclog "github.com/hashicorp/go-hclog"
 	metrics "github.com/hashicorp/go-metrics"
+	"github.com/hashicorp/nomad-autoscaler/plugins"
 	"github.com/hashicorp/nomad-autoscaler/plugins/apm"
 	"github.com/hashicorp/nomad-autoscaler/plugins/strategy"
 	"github.com/hashicorp/nomad-autoscaler/sdk"
@@ -265,9 +266,14 @@ func (ch *checkRunner) runCheckAndCapCount(ctx context.Context, currentCount int
 		return sdk.ScalingAction{}, nil
 	}
 
-	metrics, err := ch.queryMetrics(ctx, cache)
-	if err != nil {
-		return sdk.ScalingAction{}, fmt.Errorf("failed to query source: %v", err)
+	metrics := sdk.TimestampedMetrics{}
+	// Fixed-value strategy does not require APM metrics; all other strategies still do.
+	if ch.check.Strategy.Name != plugins.InternalStrategyFixedValue {
+		var err error
+		metrics, err = ch.queryMetrics(ctx, cache)
+		if err != nil {
+			return sdk.ScalingAction{}, fmt.Errorf("failed to query source: %v", err)
+		}
 	}
 
 	action, err := ch.getNewCountFromStrategy(ctx, currentCount, metrics)

--- a/policy/check_test.go
+++ b/policy/check_test.go
@@ -428,10 +428,6 @@ func TestCheckHandler_runCheckAndCapCount_IgnoredStrategyErrorsContinueEvaluatio
 }
 
 func TestCheckHandler_runCheckAndCapCount_FixedValueSkipsAPMQuery(t *testing.T) {
-	nowFunc = func() time.Time {
-		return time.Time{}
-	}
-
 	sr := &mockStrategyRunner{t: t, count: 7, direction: sdk.ScaleDirectionUp}
 	ml := &mockAPMLooker{t: t, query: "query", err: testErr}
 
@@ -441,9 +437,8 @@ func TestCheckHandler_runCheckAndCapCount_FixedValueSkipsAPMQuery(t *testing.T) 
 		MetricsGetter:  ml,
 		Policy:         testPolicy,
 	}, &sdk.ScalingPolicyCheck{
-		Name:        "fixed-value-check",
-		Source:      "mock",
-		Query:       "query",
+		Name: "fixed-value-check",
+		// source and query are not needed for fixed-value, because the APM query should be skipped entirely.
 		QueryWindow: time.Minute,
 		Strategy: &sdk.ScalingPolicyStrategy{
 			Name: plugins.InternalStrategyFixedValue,
@@ -452,8 +447,8 @@ func TestCheckHandler_runCheckAndCapCount_FixedValueSkipsAPMQuery(t *testing.T) 
 
 	action, err := runner.runCheckAndCapCount(context.Background(), 5, newQueryMetricsCache())
 	must.NoError(t, err)
-	must.Eq(t, int64(7), action.Count)
-	must.Eq(t, sdk.ScaleDirectionUp, action.Direction)
-	must.Eq(t, 0, ml.queryCalls)
-	must.Eq(t, 1, sr.runCalls)
+	test.Eq(t, int64(7), action.Count)
+	test.Eq(t, sdk.ScaleDirectionUp, action.Direction)
+	test.Eq(t, 0, ml.queryCalls, test.Sprint("should not have made any apm queries"))
+	test.Eq(t, 1, sr.runCalls)
 }

--- a/policy/check_test.go
+++ b/policy/check_test.go
@@ -10,6 +10,7 @@ import (
 	"time"
 
 	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad-autoscaler/plugins"
 	"github.com/hashicorp/nomad-autoscaler/sdk"
 	"github.com/shoenig/test"
 	"github.com/shoenig/test/must"
@@ -351,4 +352,35 @@ func TestCheckHandler_runCheckAndCapCount_OutsideSchedule(t *testing.T) {
 	must.Eq(t, sdk.ScalingAction{}, action, errMsg)
 	must.Eq(t, 0, ml.queryCalls, errMsg)
 	must.Eq(t, 0, sr.runCalls, errMsg)
+}
+
+func TestCheckHandler_runCheckAndCapCount_FixedValueSkipsAPMQuery(t *testing.T) {
+	nowFunc = func() time.Time {
+		return time.Time{}
+	}
+
+	sr := &mockStrategyRunner{t: t, count: 7, direction: sdk.ScaleDirectionUp}
+	ml := &mockAPMLooker{t: t, query: "query", err: testErr}
+
+	runner := newCheckRunner(&CheckRunnerConfig{
+		Log:            hclog.NewNullLogger(),
+		StrategyRunner: sr,
+		MetricsGetter:  ml,
+		Policy:         testPolicy,
+	}, &sdk.ScalingPolicyCheck{
+		Name:        "fixed-value-check",
+		Source:      "mock",
+		Query:       "query",
+		QueryWindow: time.Minute,
+		Strategy: &sdk.ScalingPolicyStrategy{
+			Name: plugins.InternalStrategyFixedValue,
+		},
+	})
+
+	action, err := runner.runCheckAndCapCount(context.Background(), 5, newQueryMetricsCache())
+	must.NoError(t, err)
+	must.Eq(t, int64(7), action.Count)
+	must.Eq(t, sdk.ScaleDirectionUp, action.Direction)
+	must.Eq(t, 0, ml.queryCalls)
+	must.Eq(t, 1, sr.runCalls)
 }

--- a/policy/handler.go
+++ b/policy/handler.go
@@ -14,6 +14,7 @@ import (
 
 	hclog "github.com/hashicorp/go-hclog"
 	metrics "github.com/hashicorp/go-metrics"
+	"github.com/hashicorp/nomad-autoscaler/plugins"
 	"github.com/hashicorp/nomad-autoscaler/plugins/apm"
 	"github.com/hashicorp/nomad-autoscaler/plugins/strategy"
 	targetpkg "github.com/hashicorp/nomad-autoscaler/plugins/target"
@@ -228,21 +229,34 @@ func (h *Handler) loadCheckRunners(policy *sdk.ScalingPolicy) ([]checker, error)
 
 	switch policy.Type {
 	case sdk.ScalingPolicyTypeCluster, sdk.ScalingPolicyTypeHorizontal:
-		for _, check := range policy.Checks {
-
-			s, err := h.pm.GetStrategyRunner(check.Strategy.Name)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get strategy %s: %w", check.Strategy.Name, err)
+		for i, check := range policy.Checks {
+			if check == nil {
+				return nil, fmt.Errorf("invalid check at index %d: check cannot be nil", i)
 			}
 
-			mg, err := h.pm.GetAPMLooker(check.Source)
+			if check.Strategy == nil || check.Strategy.Name == "" {
+				return nil, fmt.Errorf("invalid check %q: missing strategy value", check.Name)
+			}
+
+			strategyName := check.Strategy.Name
+
+			s, err := h.pm.GetStrategyRunner(strategyName)
 			if err != nil {
-				return nil, fmt.Errorf("failed to get APM for strategy %s: %w", check.Strategy.Name, err)
+				return nil, fmt.Errorf("failed to get strategy %s: %w", strategyName, err)
+			}
+
+			var mg apm.Looker
+			// Fixed-value strategy does not need APM; keep looker nil and skip APM lookup.
+			if strategyName != plugins.InternalStrategyFixedValue {
+				mg, err = h.pm.GetAPMLooker(check.Source)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get APM for strategy %s: %w", strategyName, err)
+				}
 			}
 
 			runner := newCheckRunner(&CheckRunnerConfig{
 				Log: h.log.Named("check_handler").With("check", check.Name,
-					"source", check.Source, "strategy", check.Strategy.Name),
+					"source", check.Source, "strategy", strategyName),
 				StrategyRunner: s,
 				MetricsGetter:  mg,
 				Policy:         policy,

--- a/policy/handler_test.go
+++ b/policy/handler_test.go
@@ -13,6 +13,7 @@ import (
 	"time"
 
 	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad-autoscaler/plugins"
 	"github.com/hashicorp/nomad-autoscaler/sdk"
 	"github.com/shoenig/test"
 	"github.com/shoenig/test/must"
@@ -461,6 +462,39 @@ func TestHandler_Run_PolicyOutsideSchedule_Integration(t *testing.T) {
 			t.Errorf("expected only 'outside schedule window' logs, got: %q", line)
 		}
 	}
+}
+
+func TestHandler_applyPolicyState_FixedValueDoesNotRequireAPM(t *testing.T) {
+	fixedValuePolicy := &sdk.ScalingPolicy{
+		ID:                 "fixed-value-policy",
+		Type:               sdk.ScalingPolicyTypeHorizontal,
+		EvaluationInterval: time.Second,
+		Target:             &sdk.ScalingPolicyTarget{Name: "mock-target", Config: map[string]string{}},
+		Checks: []*sdk.ScalingPolicyCheck{
+			{
+				Name:   "check-fixed-value",
+				Source: "missing-apm",
+				Strategy: &sdk.ScalingPolicyStrategy{
+					Name:   plugins.InternalStrategyFixedValue,
+					Config: map[string]string{"value": "3"},
+				},
+			},
+		},
+	}
+
+	h := &Handler{
+		log: hclog.NewNullLogger(),
+		pm: &MockDependencyGetter{
+			APMLookerErr: testErr,
+			StrategyRunner: &mockStrategyRunner{
+				t: t,
+			},
+		},
+	}
+
+	err := h.applyPolicyState(fixedValuePolicy)
+	must.NoError(t, err)
+	must.Eq(t, 1, len(h.checkRunners))
 }
 
 var policy = &sdk.ScalingPolicy{

--- a/policy/handler_test.go
+++ b/policy/handler_test.go
@@ -472,8 +472,8 @@ func TestHandler_applyPolicyState_FixedValueDoesNotRequireAPM(t *testing.T) {
 		Target:             &sdk.ScalingPolicyTarget{Name: "mock-target", Config: map[string]string{}},
 		Checks: []*sdk.ScalingPolicyCheck{
 			{
-				Name:   "check-fixed-value",
-				Source: "missing-apm",
+				Name: "check-fixed-value",
+				// source and query are not needed for fixed-value, because the APM query should be skipped entirely.
 				Strategy: &sdk.ScalingPolicyStrategy{
 					Name:   plugins.InternalStrategyFixedValue,
 					Config: map[string]string{"value": "3"},

